### ISSUE TITLE
CLI: Generate a new CLI jar each time a new code gets merged

### ISSUE THREAD

**Issue by nollymar** [dotCMS/core#27286]
### Task

We have a GHA to [generate CLI releases](https://github.com/dotCMS/core/actions/workflows/cli-release-process.yml). However, this task is performed manually. 

We have to automate it to generate a CLI release each time a new code from the CLI source code gets merged. Once the snapshot version is published, it should be tagged as a release candidate (alpha, beta, etc ) version in NPM, so the QA team can install it easily.

### Proposed Objective

Core Features

### Proposed Priority

Priority 3 - Average

**Comment by nollymar** [comment]
**Internal QA:** Needs work

The jar is generated on each merge but the jar version is always 1.0.0-SNAPSHOT. We should find a way to include a suffix such as hash commit or something similar to make the jar unique.

**Comment by nollymar** [comment]
**Internal QA:** Needs work

Apparently, the changes made on this card broke the quarkus execution https://dotcms.slack.com/archives/C06M2Q9PY7R/p1711389553614169

**Comment by fabrizzio-dotCMS** [comment]
quite a few things to notice here :
The jar isn't getting published here after the last merge from yesterday https://repo.dotcms.com/artifactory/libs-snapshot-local/com/dotcms/dotcms-cli/

The info on the MANIFEST.MF of the uber jar looks exactly like this:

```
Manifest-Version: 1.0
Implementation-Build: 6abb714
Implementation-Timestamp: 1712675889180
Implementation-URL: https://github.com/dotCMS/dotcms-cli-parent/dotcms-c
 li
Java-Vendor: Eclipse Adoptium
Java-Version: 11
Os-Arch: aarch64
Os-Name: Mac OS X
Os-Version: 14.4.1
Scm-Build: issue-27386-limited-user-folder-checks
Scm-Connection: scm:git:git@github.com:dotCMS/core.git
Scm-Revision: 6abb714
Scm-Url: https://github.com/dotCMS/core
Class-Path: 
Main-Class: io.quarkus.runner.GeneratedMain
Implementation-Title: dotcms-cli
Implementation-Version: 1.0.0-SNAPSHOT
Multi-Release: true
```
I think we need to revise the URL on Implementation-URL
See it is cut and it leads to 404

On the brighter side. I tested the command line executing the quarkus dev command and it is once again back to normal.

**Comment by dcolina** [comment]
It will be fix into #28226

**Comment by nollymar** [comment]
**Internal QA:** Passed

A new CLI jar is generated on [here](https://repo.dotcms.com/ui/native/libs-snapshot-local/com/dotcms/dotcms-cli/1.0.0-SNAPSHOT/) each time a new code gets merged into master
